wrappers/rust/icicle-cuda-runtime/src/memory.rs

```diff
@@ -117,6 +134,66 @@ impl<T> HostSlice<T> {
         self.0
             .iter_mut()
     }
+
+    pub fn is_pinnable(&self) -> bool {
+        let pinnable = get_device_attribute(cudaDeviceAttr::cudaDevAttrHostRegisterSupported, 0).unwrap();
+        let lockable =
+            get_device_attribute(cudaDeviceAttr::cudaDevAttrPageableMemoryAccessUsesHostPageTables, 0).unwrap();
+
+        pinnable == 1 && lockable == 0
+    }
+
+    pub fn pin(&self, flags: CudaHostRegisterFlags) -> CudaResult<()> {
+        unsafe {
+            let ptr = self.as_ptr() as *mut c_void;
+            let flags_to_set = flags.bits();
+            cudaHostRegister(ptr, self.len(), flags_to_set as c_uint).wrap()
```

Comment:
Removed for now. cudaHostRegister would fail if the following was false (0). 
```rust
pinnable = get_device_attribute(cudaDeviceAttr::cudaDevAttrHostRegisterSupported, 0).unwrap();
```